### PR TITLE
Drop support for `.NETCore3.1` in Tests and samples

### PR DESCRIPTION
**Pull request by eaba**
## Changes

Drop support for .NET Core 3.1

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [x] Design discussion issue https://github.com/akkadotnet/akka.net/discussions/6527

### CODE REVIEW

File: build.fsx [hunk 1]
```diff
@@ -56,7 +56,6 @@ let incrementalistReport = output @@ "incrementalist.txt"
 
 // Configuration values for tests
 let testNetFrameworkVersion = "net471"
-let testNetCoreVersion = "netcoreapp3.1"
```
Comment (eaba) [inline build.fsx:59]: I remove "netcoreapp3.1" from `build.fsx` - everything.

File: src/benchmark/Akka.Benchmarks/Akka.Benchmarks.csproj [hunk 2]
```diff
@@ -3,7 +3,7 @@
   <Import Project="..\..\common.props" />
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFrameworks>$(NetTestVersion);$(NetCoreTestVersion)</TargetFrameworks>
```
Comment (eaba) [inline src/benchmark/Akka.Benchmarks/Akka.Benchmarks.csproj:6]: I only removed this everywhere........

**Review by eaba — COMMENTED** [review]
please check it @Aaronontheweb

File: src/core/Akka.Tests/Akka.Tests.csproj [hunk 3]
```diff
@@ -4,8 +4,8 @@
 
   <PropertyGroup>
     <AssemblyTitle>Akka.Tests</AssemblyTitle>
-    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">$(NetFrameworkTestVersion);$(NetTestVersion);$(NetCoreTestVersion)</TargetFrameworks>
-    <TargetFrameworks Condition=" '$(OS)' != 'Windows_NT' ">$(NetTestVersion);$(NetCoreTestVersion)</TargetFrameworks>
+    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">$(NetFrameworkTestVersion);$(NetTestVersion)</TargetFrameworks>
```
Comment (Aaronontheweb) [inline src/core/Akka.Tests/Akka.Tests.csproj:7]: **In a different pull request**, we can probably get rid of these conditionals - I believe they were added many years ago for working around issues with Mono execution.

**Review by Aaronontheweb — APPROVED** [review]
LGTM - just waiting to see if the same two tests on .NET 7 fail again after this.

File: src/core/Akka.Tests/Akka.Tests.csproj (hunk 3 above)
Comment (eaba) [reply src/core/Akka.Tests/Akka.Tests.csproj:7]: ok

**Pull request merged**